dotnet-client-libraries/BenchmarkDotNetWithUnitTests/Tests/StringConcatBenchmarkLiveTest.cs

```diff
@@ -0,0 +1,56 @@
+using System.Collections.Immutable;
+using BenchmarkDotNet.Reports;
+
+namespace Tests;
+
+public class StringConcatBenchmarkLiveTest : IClassFixture<BenchmarkFixture>
+{
+    private readonly ImmutableArray<BenchmarkReport> _benchmarkReports;
+
+    public StringConcatBenchmarkLiveTest(BenchmarkFixture benchmarkFixture)
+    {
+        _benchmarkReports = benchmarkFixture.BenchmarkSummary.Reports;
+    }
+
+    [Fact]
+    public void WhenBenchmarkTestsAreRun_ThenFourCasesMustBeExecuted()
+    {
+        var benchmarkCases = _benchmarkReports.Length;
+        Assert.Equal(4, benchmarkCases);
+    }
+
+
+    [Fact]
+    public void WhenStringInterpolationCaseIsExecuted_ThenItShouldNotTakeMoreThanFifteenNanoSecs()
+    {
+        var stats = _benchmarkReports.First(x =>
+            x.BenchmarkCase.Descriptor.DisplayInfo == "StringConcatBenchmarks.StringInterpolation").ResultStatistics;
```

Comment:
@AndreCarvalho  Done

Reply:
Just as a side note - not disagreeing with the change, but wanted to be sure it is clear that now instead of only doing this operation in a couple of tests, it will be done for each test. xUnit runs the constructor for each test that is executed. This means that the call to `First` will be executed for each test. If we are looking at this change for a performance gain, then it should be moved into the `ClassFixture`, which will execute once for all tests within this class.

Reply:
@jshergal @AndreCarvalho, It doesn't have any impact. The execution speed remains the same at the ctor or the method level with this change. I prefer to keep the root level object `BenchmarkSummary` at the `ClassFixture` level only as it allows the developer to extract any information in the `TestClass` and write different test methods based on it. As this code was explicitly focussing on the `StringInterpolation,` I condensed those changes at the constructor level of the `TestClass.`

Reply:
@goldytech - No worries. I would not expect it to really have any real impact based on the current tests (just a handful). I was pointing it out more from the technical side of things, as some don't understand or realize that xUnit creates one class instance for each test.